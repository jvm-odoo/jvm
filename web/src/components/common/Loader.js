import React from 'react'
import styled from 'styled-components'

import loaderImage from '../../assets/img/loader.svg'

const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const LoaderImage = styled.img`
  height: 64px;
  width: 64px;
`

export const Loader = () => (
  <LoaderContainer>
    <LoaderImage src={loaderImage} alt="Loading..." />
  </LoaderContainer>
)
