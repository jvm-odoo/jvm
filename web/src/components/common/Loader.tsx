import React from 'react'
import styled from 'styled-components'

import loaderImage from '../../assets/img/loader.svg'

const LoaderContainer = styled.div<ILoaderProps>`
    display: flex;
    justify-content: center;
    align-items: center;
    ${({ full }) =>
        full &&
        `
    height: 100vh;
    width: 100vw;
  `}
`

const LoaderImage = styled.img.attrs(() => ({
    loading: 'lazy',
    decoding: 'async',
}))`
    height: 64px;
    width: 64px;
`

interface ILoaderProps {
    full?: boolean
}

export const Loader = ({ full }: ILoaderProps) => (
    <LoaderContainer full={full}>
        <LoaderImage src={loaderImage} alt="Loading..." />
    </LoaderContainer>
)
