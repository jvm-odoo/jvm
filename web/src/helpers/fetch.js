export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
