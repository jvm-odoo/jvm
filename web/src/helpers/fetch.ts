export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}