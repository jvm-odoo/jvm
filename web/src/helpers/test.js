import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components'

import { store } from '../redux/store'
import { THEME_LIGHT } from '../constants'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(
    <ThemeProvider theme={THEME_LIGHT}>
      {Component}
    </ThemeProvider>)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ThemeProvider theme={THEME_LIGHT}>
          {Component}
        </ThemeProvider>
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}

export const withReduxProvider = (Component) => (
  <Provider store={store}>
    {Component}
  </Provider>
)
