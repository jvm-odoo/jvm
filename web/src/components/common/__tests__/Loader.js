import React from 'react'

import { Loader } from '../Loader'
import { shallowMatchSnapshot } from '../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<Loader />))
