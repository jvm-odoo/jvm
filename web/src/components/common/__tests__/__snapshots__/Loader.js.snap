// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.img
  alt="Loading..."
  src="loader.svg"
/>
`;
